### ISSUE TITLE
请问有谁复现出SST-2的结果了吗，作者能提供下参数吗

### ISSUE THREAD

**Issue by psy2013GitHub** [Sanyuan-Chen/RecAdam#7]

**Comment by Sanyuan-Chen** [comment]
你好~
我们在实验中得到的SST-2数据集上的median dev score 93.6对应的超参数为: recadam_anneal_t0=1000, recadam_anneal_k=1，其它的超参数请详见Readme的介绍。 
同时，我们也发现有很多组超参数也可以得到类似的实验结果，如：当recadam_anneal_t0=1000, recadam_anneal_k=0.2, median dev score=93.5; 当recadam_anneal_t0=250, recadam_anneal_k=0.2, median dev score=93.5; 
由于实验环境和随机种子的差别，可能复现得到的实验结果会有略微的区别。请问你那边复现时使用的超参数和得到的结果分别是多少呢？

**Comment by psy2013GitHub** [comment]
你好，感谢回复～如下参数92%
export GLUE_DIR=./glue_data/
export TASK_NAME=SST-2

export OUTPUT_DIR=/data/$TASK_NAME/RecAdam`

python run_glue_with_RecAdam.py \
  --model_type bert \
  --model_name_or_path bert-base-uncased \
  --log_path /data/tmp/logs/adam.log \
  --task_name $TASK_NAME \
  --do_eval \
  --do_lower_case \
  --data_dir $GLUE_DIR/$TASK_NAME \
  --max_seq_length 128 \
  --learning_rate 5e-5 \
  --max_steps 21050 \
  --evaluate_during_training \
  --train_logging_steps 50 \
  --eval_logging_steps 180 \
  --recadam_anneal_fun sigmoid \
  --recadam_anneal_t0 1000 \
  --recadam_anneal_k 1 \
  --recadam_pretrain_cof 5000.0 \
  --logging_Euclid_dist \
  --save_steps 100 \
  --overwrite_output_dir \
  --per_gpu_train_batch_size 48 \
  --per_gpu_eval_batch_size 128 \
  --eval_all_checkpoints \
  --output_dir ${OUTPUT_DIR} \
  --optimizer RecAdam \
  --do_train \
  --warmup_steps 2000 \
  --weight_decay 0.01 \`

环境信息：
python 3.6.9
torch 1.5
单卡1080ti

**Comment by psy2013GitHub** [comment]
学习率改成2e-5就可以复现了，请问学习率影响很大吗，是不是即使5e-5，调节其他参数也能93.5 — —

**Comment by Sanyuan-Chen** [comment]
你好~
我们在实验中使用的learning_rate, max_seq_length, batch_size等超参数是直接沿用[Huggingface example](https://github.com/huggingface/transformers/tree/c44a17db1b87e31ad4c232e48d19a2700e8b690d/examples)。
为了将我们的方法和baseline进行公平地比较，我们只对于我们RecAdam相关的超参数recadam_anneal_t0，recadam_anneal_k进行了调节，而没有去尝试调节learning_rate等原有的超参数，所以也没有相应的实验结果可以提供
如果感兴趣的话，欢迎fork并进行一些尝试~
另外，近期也有一些工作提出通过使用较小的learning rate和更多的training iteration可以得到更稳定且有效的结果：[On the Stability of Fine-tuning BERT: Misconceptions, Explanations, and Strong Baselines](https://www.lsv.uni-saarland.de/wp-content/publications/2020/On_the_Stability_of_Fine-tuning_BERT_preprint.pdf) ([twitter version](https://twitter.com/mariusmosbach/status/1270387450188365826)), [Revisiting Few-sample BERT Fine-tuning
](https://arxiv.org/pdf/2006.05987.pdf) ([twitter version](https://twitter.com/yoavartzi/status/1271089161844424704))。如果感兴趣可以了解一下~

**Comment by psy2013GitHub** [comment]
> 你好~
> 我们在实验中使用的learning_rate, max_seq_length, batch_size等超参数是直接沿用[Huggingface example](https://github.com/huggingface/transformers/tree/c44a17db1b87e31ad4c232e48d19a2700e8b690d/examples)。
> 为了将我们的方法和baseline进行公平地比较，我们只对于我们RecAdam相关的超参数recadam_anneal_t0，recadam_anneal_k进行了调节，而没有去尝试调节learning_rate等原有的超参数，所以也没有相应的实验结果可以提供
> 如果感兴趣的话，欢迎fork并进行一些尝试~
> 另外，近期也有一些工作提出通过使用较小的learning rate和更多的training iteration可以得到更稳定且有效的结果：[On the Stability of Fine-tuning BERT: Misconceptions, Explanations, and Strong Baselines](https://www.lsv.uni-saarland.de/wp-content/publications/2020/On_the_Stability_of_Fine-tuning_BERT_preprint.pdf) ([twitter version](https://twitter.com/mariusmosbach/status/1270387450188365826)), [Revisiting Few-sample BERT Fine-tuning ](https://arxiv.org/pdf/2006.05987.pdf) ([twitter version](https://twitter.com/yoavartzi/status/1271089161844424704))。如果感兴趣可以了解一下~

谢谢Sanyuan~
